core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -129,12 +131,23 @@ public static FileObject getFileObject( String vfsFilename, VariableSpace space,
       // If not, we are going to assume it's a file.
       //
       boolean relativeFilename = true;
-      String[] schemes = fsManager.getSchemes();
-      for ( int i = 0; i < schemes.length && relativeFilename; i++ ) {
-        if ( vfsFilename.startsWith( schemes[ i ] + ":" ) ) {
+      String[] initialSchemes = fsManager.getSchemes();
+
+      relativeFilename = checkForScheme( initialSchemes, relativeFilename, vfsFilename, space, fsOptions );
+
+      int timeOut = timeOutLimit;
+      boolean hasScheme = vfsFilename != null && vfsFilename.contains( "://" ); // check for bigData providers
+
+      while ( relativeFilename && hasScheme && timeOut > 0 ) {
+        String[] schemes = fsManager.getSchemes();
+        try {
+          Thread.sleep( timeToSleepStep );
+          timeOut -= timeToSleepStep;
+          relativeFilename = checkForScheme( schemes, relativeFilename, vfsFilename, space, fsOptions );
```

Comment:
Could this while loop just check repeatedly for whether the scheme is available from fsManager.getSchemes()?

The .checkForSchemes() seems to be doing too much to me-- looping each time to find whether or not the scheme is present, and then calling buildFsOptions.